core/bin/via_server/src/node_builder.rs

```diff
@@ -367,7 +377,7 @@ impl ViaNodeBuilder {
             .add_circuit_breaker_checker_layer()?
             .add_postgres_metrics_layer()?
             .add_query_eth_client_layer()?
-            .add_storage_initialization_layer(LayerKind::Task)?
+            .add_storage_initialization_layer(LayerKind::Precondition)?
```

Comment:
why do we need these layers twice here?

Reply:
1. The precondition version (called earlier in the build process) ensures that the storage is in a valid state before other components of the node are initialized. This might involve checking for the presence of genesis data or other critical information.
2- The task version (called later in the build process) might perform more extensive initialization or ongoing maintenance of the storage. This could include tasks like applying updates, performing migrations, or initializing components that depend on a fully set up storage system.

and also this is the same behavior that the zksync node builder apply to initializing their node.